documentation/api/addAssertion.md

```diff
@@ -10,82 +10,234 @@ expect.addAssertion(pattern, handler);
 expect.addAssertion([pattern, ...]], handler);
 ```
 
-New assertions can be added to Unexpected the following way.
+`expect.addAssertion` takes two arguments:
+
+1. a string pattern (or an array of patterns) that describes the assertion.
+2. a handler function that is called when when the assertion is invoked.
+
+For example:
+
+```js
+function handler(expect, subject, value) {
```

Comment:
I think I'd suggest we inline the handler function here. The reasoning is almost no assertions in core or any of the plugins separate the function out like this, and I think it would be good to show this example exactly as it is in the wild. So perhaps arrow function?

The other reason is that we probably want to have a `return` in front of the inner assertion. There is no semantic difference if you are deferring to a synchronous assertion, but things will just not work right if you forget it with "to satisfy" for example and it goes async at some future point.

Reply:
I agree but I defined it out there so I could reference it in the examples that follow. Of course I could still inline (and name it) and reference it. What do you think?

About `return`, I only thought to follow the design of the previous examples. And there's a section about async assertions down the page. Or is the idea to always use `return` in examples going forward?

Reply:
I have to say I don’t feel too strongly about the return, was just an idle thought about a possible pitfall.

But I do think the inlining is important. Let me do a top to bottom read and see if I can improve the suggestion so it works contextually.